src/CloudFoundry.Test.Integration/NatsClientTest.cs

```diff
@@ -304,8 +304,11 @@ public void TC012_RecieveMessageOnUnsubscription()
             {
                 natsClient.Publish("foo", () =>
                 {
+                    mre.Reset();
```

Comment:
Is this reset necessary? 
It may cause a deadlock if the execution is in the following order:
1. mre.Set();
2. mre.Reset();
3. mre.WaitOne();